extensions/ql-vscode/src/telemetry.ts

```diff
@@ -209,6 +209,7 @@ export let telemetryListener: TelemetryListener;
 
 export async function initializeTelemetry(extension: Extension<any>, ctx: ExtensionContext): Promise<void> {
   telemetryListener = new TelemetryListener(extension.id, extension.packageJSON.version, key, ctx);
-  telemetryListener.initialize();
+  // do not await initialization, since doing so will cause integration tests to hang.
+  void telemetryListener.initialize();
```

Comment:
Huh, strange that this is the case. Do you know why the integration tests hang if we wait for the telemetry to initialize? In any case this seems okay to me, since we weren't `await`ing it before so this isn't a behaviour change.

Reply:
It's because in some situations, the initialize function can launch a dialog box to ask if a user wants to enable telemetry. The function blocks until the dialog closes. There is no programmatic way (AFAIK) to close a dialog.

Reply:
Aha, makes sense. Could you expand the comment a little bit to explain that to future readers of the code?